src/core/server/services/users/users.ts

```diff
@@ -682,21 +772,38 @@ export async function updateModerationScopes(
     });
   }
 
-  if (user.id === userID) {
+  // you can only change scopes if you're currently an admin or moderator
+  if (![GQLUSER_ROLE.ADMIN, GQLUSER_ROLE.MODERATOR].includes(viewer.role)) {
+    throw new Error("you have insufficient permissions to promote users");
+  }
+
+  // scoped moderators (site mods) should not be using updateModerationScopes
+  // they should be using promoteUserRole
+  if (
+    viewer.role === GQLUSER_ROLE.MODERATOR &&
+    viewer.moderationScopes &&
+    viewer.moderationScopes.siteIDs?.length !== 0
+  ) {
+    throw new Error("you have insufficient permissions to set roles");
+  }
```

Comment:
This check should be completed at the schema level and should not be duplicated here.